src/client.ts

```diff
@@ -837,39 +837,8 @@ export class MatrixClient extends EventEmitter {
         // actions for themselves, so we have to kinda help them out when they are encrypted.
         // We do this so that push rules are correctly executed on events in their decrypted
         // state, such as highlights when the user's name is mentioned.
-        this.on("Event.decrypted", (event) => {
-            const oldActions = event.getPushActions();
-            const actions = this.pushProcessor.actionsForEvent(event);
-            event.setPushActions(actions); // Might as well while we're here
-
-            const room = this.getRoom(event.getRoomId());
-            if (!room) return;
-
-            const currentCount = room.getUnreadNotificationCount(NotificationCountType.Highlight);
-
-            // Ensure the unread counts are kept up to date if the event is encrypted
-            // We also want to make sure that the notification count goes up if we already
-            // have encrypted events to avoid other code from resetting 'highlight' to zero.
-            const oldHighlight = oldActions && oldActions.tweaks
-                ? !!oldActions.tweaks.highlight : false;
-            const newHighlight = actions && actions.tweaks
-                ? !!actions.tweaks.highlight : false;
-            if (oldHighlight !== newHighlight || currentCount > 0) {
-                // TODO: Handle mentions received while the client is offline
-                // See also https://github.com/vector-im/element-web/issues/9069
-                if (!room.hasUserReadEvent(this.getUserId(), event.getId())) {
-                    let newCount = currentCount;
-                    if (newHighlight && !oldHighlight) newCount++;
-                    if (!newHighlight && oldHighlight) newCount--;
```

Comment:
I swear there was a reason why we did a `-1` in addition to `+1`, but clearly 2019 me didn't bother to document it. Not really worth exploring in this PR, but something to watch for after this lands: it'll probably be some subtle bug that has been supressed from my nightmares for 2 years.

Reply:
Though, a quick manual test of scrolling up in an encrypted room with messages sitting in that backlog would be appreciated as I think it might be for that?

Reply:
Scrolling through an encrypted room isn't causing any weirdness for me. I am also a bit scared about what this could cause. Also, I am away next week so if this sets Element on fire I won't be there to fix it :sweat_smile: So feel free to delay this or whatever you find suitable